src/Transport/Receiving/MessageReceiverNotifier.cs

```diff
@@ -87,6 +91,14 @@ await Task.WhenAll(tasks)
         }
 
         public async Task Stop()
+        {
+            if (Interlocked.Decrement(ref refCount) == 0)
+            {
+                await StopInternal().ConfigureAwait(false);
```

Comment:
Is it possible that this could be stopped more times than it is started? Afterwards you would have to call Start() enough times to bring it back to zero before it would start again.

Reply:
As of now we don't support restarting the same endpoint plus didn't the previous code have exactly the same assumptions? Do you think this is an issue we should address?

Reply:
I agree with this argument. This works better than the previous because multiple `Stop()` calls won't try to shut the thing down after it's already shut down.

Reply:
similar to mike's comment: is it guaranteed that `Stop` is called at least as many times as start? Otherwise a dedicated counter might be safer?